SYCL/DeviceLib/built-ins/fast-math-flag.cpp

```diff
@@ -0,0 +1,82 @@
+// RUN: %clangxx -fsycl -ffast-math -fsycl-targets=%sycl_triple %s -o %t.out
+// RUN: %HOST_RUN_PLACEHOLDER %t.out
+// RUN: %CPU_RUN_PLACEHOLDER %t.out
+// RUN: %GPU_RUN_PLACEHOLDER %t.out
+// RUN: %ACC_RUN_PLACEHOLDER %t.out
+
+#include <CL/sycl.hpp>
+#include <cassert>
+
+#define __TEST_FFMATH_BINARY(func)                                             \
+  int test_ffmath_##func() {                                                   \
+    sycl::float4 r[2];                                                         \
+    sycl::float4 val[2] = {{1.0004f, 1e-4f, 1.4f, 14.0f},                      \
+                           {1.0004f, 1e-4f, 1.4f, 14.0f}};                     \
+    {                                                                          \
+      sycl::buffer<sycl::float4, 1> output(&r[0], sycl::range<1>(2));          \
+      sycl::buffer<sycl::float4, 1> input(&val[0], sycl::range<1>(2));         \
+      sycl::queue q;                                                           \
+      q.submit([&](sycl::handler &cgh) {                                       \
+        auto AccO =                                                            \
+            output.template get_access<sycl::access::mode::write>(cgh);        \
+        auto AccI = input.template get_access<sycl::access::mode::read>(cgh);  \
+        cgh.single_task([=]() {                                                \
+          AccO[0] = sycl::func(AccI[0], AccI[1]);                              \
+          AccO[1] = sycl::native::func(AccI[0], AccI[1]);                      \
+        });                                                                    \
+      });                                                                      \
+    }                                                                          \
+    return sycl::all(r[0] == r[1]);                                            \
```

Comment:
I can't get purpose of `sycl::all` for bool converted for integer here.
What's purpose of the test?

Reply:
The purpose of the test is to check if all the vector components are equal implying that the `::native` function and the regular one return the same values.

Let me know if it is fine, otherwise I can change it.

Reply:
Per SYCL 2020 spec, `all(igeninter)` `Returns 1 if the most significant bit in all components of x is set; otherwise returns 0.`
On line 29 a boolean is provided instead.

Reply:
Thanks for point this out, I misunderstood the initial message.

I think that the `operator==` returns a signed integer vector of size 4, in which each component is equal to -1 (all bits are set) if the corresponding values of `r[0]` and `r[1]` have the same value, 0 otherwise (no bit is set). For reference, see [Table 141](https://www.khronos.org/registry/SYCL/specs/sycl-2020/html/sycl-2020.html#_vec_interface).

Reply:
For char, short, int and long it's cahr short int and long respectively. For any other type it's a mere boolean.

Reply:
It depends on the size of the type. So, in the case of floats the returned vector type is made of `cl_int` as mandated by the SYCL specifications linked in my previous message.